test/functional/test_framework/test_framework.py

```diff
@@ -430,11 +430,12 @@ def _initialize_chain(self):
             self.stop_nodes()
             self.nodes = []
             self.disable_mocktime()
+            def cache_path(*paths):
+                return os.path.join(get_datadir_path(self.options.cachedir, i), "regtest", *paths)
```

Comment:
`i` won't be in scope yet; have to declare this in the loop.

Reply:
I don't think that's necessary. Python will add the symbol i to the scope of the function, not thefor  block. Though it might be clearer to just pass `i` in as a parameter.

Reply:
Oh, @laanwj is right. Still probably best to either parameterize or move this definition as-is under the `for` for clarity.

Reply:
Not that the performance overhead is too bad in this case, but I think defining a function inside a loop is generally frowned upon. AFAIK essentially it's instantiating an object every time.